dnf-behave-tests/features/repoquery/deps.feature

```diff
@@ -0,0 +1,559 @@
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom1-prov1
+      bottom1-prov2 = 1.0
+      bottom1-prov3 <= 1.2
+      bottom1-prov3 > 1.5
+      bottom2 = 1:1.0-1
+      bottom3
+      bottom3-prov1
+      """
+
+Scenario: repoquery --requires NAMEGLOB-VERSION NAME
+ When I execute dnf with args "repoquery --requires middle[1]-1.0 middle2"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom1-prov1
+      bottom1-prov2 = 1.0
+      bottom1-prov3 <= 1.2
+      bottom1-prov3 > 1.5
+      bottom2 = 1:1.0-1
+      bottom3
+      bottom3-prov1
+      """
+
+
+# --provides
+Scenario: repoquery --provides NAME
+ When I execute dnf with args "repoquery --provides bottom1"
+ Then the exit code is 0
+  And stdout is
+      """
+      a-common-provide
+      bottom1 = 1:1.0-1
+      bottom1 = 1:2.0-1
+      bottom1(x86-64) = 1:1.0-1
+      bottom1(x86-64) = 1:2.0-1
+      bottom1-prov1
+      bottom1-prov2 = 1.0
+      bottom1-prov2 = 2.0
+      bottom1-prov3 <= 1.3
+      """
+
+Scenario: repoquery --provides NAME-VERSION
+ When I execute dnf with args "repoquery --provides bottom1-1.0"
+ Then the exit code is 0
+  And stdout is
+      """
+      a-common-provide
+      bottom1 = 1:1.0-1
+      bottom1(x86-64) = 1:1.0-1
+      bottom1-prov1
+      bottom1-prov2 = 1.0
+      """
+
+Scenario: repoquery --provides NAMEGLOB-VERSIONGLOB
+ When I execute dnf with args "repoquery --provides bottom[2]-*"
+ Then the exit code is 0
+  And stdout is
+      """
+      a-common-provide
+      bottom2 = 1:1.0-1
+      bottom2 = 1:2.0-1
+      bottom2(x86-64) = 1:1.0-1
+      bottom2(x86-64) = 1:2.0-1
+      bottom2-prov1
+      """
+
+
+# --conflicts
+Scenario: repoquery --conflicts NAME
+ When I execute dnf with args "repoquery --conflicts bottom2"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom1
+      bottom3 < 2.0
+      bottom3 = 1:1.0-1
+      """
+
+Scenario: repoquery --conflicts NAME-VERSION
+ When I execute dnf with args "repoquery --conflicts bottom2-1.0"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom1
+      bottom3 = 1:1.0-1
+      """
+
+Scenario: repoquery --conflicts NAMEGLOB-VERSIONGLOB
+ When I execute dnf with args "repoquery --conflicts bottom*-?.0"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom1
+      bottom3 < 2.0
+      bottom3 = 1:1.0-1
+      """
+
+
+# --obsoletes
+Scenario: repoquery --obsoletes NAME
+ When I execute dnf with args "repoquery --obsoletes bottom3"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom1 < 7
+      bottom2
+      bottom2 = 1:2.0-1
+      """
+
+Scenario: repoquery --obsoletes NAME-VERSION
+ When I execute dnf with args "repoquery --obsoletes bottom3-1.0"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom1 < 7
+      bottom2
+      """
+
+Scenario: repoquery --obsoletes NAMEGLOB-VERSION
+ When I execute dnf with args "repoquery --obsoletes bottom[3]-1.0"
```

Comment:
If I replace argument `bottom[3]-1.0` `by bottom*-1.0`, will be there any difference in the output?

Reply:
Depends on whether more packages would match, but we do have test with `*` with e.g. the `--conflicts` switch, so having this one be different is better I think.